src/integer_traits.rs

```diff
@@ -0,0 +1,509 @@
+//! Implementation of standard traits like Add, Sub, etc. for the Integer struct
+
+use crate::{integer::IntegerType, value::private::ReprValue, Integer};
+use std::ops::{Add, Div, Mul, Sub};
+
+impl PartialEq for Integer {
+    fn eq(&self, other: &Self) -> bool {
+        match (self.integer_type(), other.integer_type()) {
+            (IntegerType::Fixnum(a), IntegerType::Fixnum(b)) => a.as_rb_value() == b.as_rb_value(),
+            (IntegerType::Bignum(a), IntegerType::Bignum(b)) => unsafe {
+                rb_sys::rb_big_eq(a.as_rb_value(), b.as_rb_value()) == rb_sys::Qtrue.into()
+            },
+            (IntegerType::Fixnum(a), IntegerType::Bignum(b)) => unsafe {
+                let a = rb_sys::rb_int2big(a.to_isize());
+                rb_sys::rb_big_eq(a, b.as_rb_value()) == rb_sys::Qtrue.into()
+            },
+            (IntegerType::Bignum(a), IntegerType::Fixnum(b)) => unsafe {
+                rb_sys::rb_big_eq(a.as_rb_value(), b.as_rb_value()) == rb_sys::Qtrue.into()
+            },
+        }
+    }
+}
+
+impl PartialEq<i64> for Integer {
+    fn eq(&self, other: &i64) -> bool {
+        self.eq(&Integer::from_i64(*other))
+    }
+}
+
+impl PartialEq<Integer> for i64 {
+    fn eq(&self, other: &Integer) -> bool {
+        Integer::from_i64(*self).eq(other)
+    }
+}
+
+impl PartialOrd for Integer {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        match (self.integer_type(), other.integer_type()) {
+            (IntegerType::Fixnum(a), IntegerType::Fixnum(b)) => {
+                (a.as_rb_value() as i64).partial_cmp(&(b.as_rb_value() as i64))
+            }
+            (IntegerType::Bignum(a), IntegerType::Bignum(b)) => unsafe {
+                let result = rb_sys::rb_big_cmp(a.as_rb_value(), b.as_rb_value());
+                Integer::from_rb_value_unchecked(result)
+                    .to_i8()
```

Comment:
I'm not sure there is a grantee that the result of `rb_big_cmp` is -1/0/+1, just that it's negative/0/positive. I'd use `to_i64` here incase it's out of the expected range (plus `to_i8` is implemented as `to_i64` plus bounds checks, so it's only ever going to be slower).